hw/top_earlgrey/syn/constraints.sdc

```diff
@@ -64,6 +68,45 @@ set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dp_en_o]        -clock ${USB_C
 set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dn_o]           -clock ${USB_CLK_PIN}
 set_output_delay ${OUT_DEL} [get_ports dio_usbdev_dn_en_o]        -clock ${USB_CLK_PIN}
 
+#####################
+# IO clk (24MHz)    #
+#####################
+set IO_CLK_PIN clk_io_i
+set IO_TCK 40.0
+set_ideal_network ${IO_CLK_PIN}
+
+create_clock ${IO_CLK_PIN} -period ${IO_TCK}
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} ${IO_CLK_PIN}
+
+set IN_DEL  IO_TCK/2
```

Comment:
For whatever it's worth, I always preferred to have these set (for each clock domain) by a percentage of clock period, something like
`set IN_DEL [expr ${IO_TCK} * IN_PCT / 100]`
`set OUT_DEL [expr ${IO_TCK} * OUT_PCT / 100]`

(I'm sure that's not 100% correct syntax, I'm very rusty)

It makes it smoother/easier to ratchet up the margin in the design.  I think you might eventually even keep an array/hash of these percentages by clock domain in case some need more margin than others.